dynamorm/table.py

```diff
@@ -773,3 +773,178 @@ def Q(**mapping):
+            self.resp = self._get_resp()
+
+            # Store the last key from query
+            self.last = self.resp.get('LastEvaluatedKey', None)
+
+        # If a Limit is specified we must not operate in recursive mode
+        if 'Limit' in self.dynamo_kwargs:
+            log.warning(
+                "%s was invoked with both a limit and the recursive flag set. "
+                "The recursive flag will be ignored",
+                self.__class__.__name__
+            )
+            self._recursive = False
+
+        # Increment which record we're going to pull from the items
+        self.index += 1
+
+        if self.index == self.resp['Count']:
+            # If we have no more items them we're done as long as we're not in recursive mode
+            # And if we are in recursive mode we're done if the resp didn't contain a last key
+            if not self._recursive or self.last is None:
+                raise StopIteration
+
+            # Our last marker is not None and we are in recursive mode
+            # Reset our response state and re-call next
+            self.again()
+            return self.__next__()
+
+        # Grab the raw item from the response and return it as a new instance of our model
+        raw = self.resp['Items'][self.index]
+        return self.model.new_from_raw(raw, partial=self._partial)
+
+    def limit(self, limit):
+        """Set the limit value"""
+        self.dynamo_kwargs['Limit'] = limit
+        return self
+
+    def start(self, last):
+        """Set the last value"""
+        self.dynamo_kwargs['ExclusiveStartKey'] = last
+        return self
+
+    def consistent(self):
+        """Make this read a consistent one"""
+        self.dynamo_kwargs['ConsistentRead'] = True
+        return self
+
+    def specific_attributes(self, attrs):
+        """Return only specific attributes in the documents through a ProjectionExpression
+
+        This is a list of attribute names. See the documentation for more info:
+        https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.ProjectionExpressions.html
+        """
+        if 'ExpressionAttributeNames' not in self.dynamo_kwargs:
+            self.dynamo_kwargs['ExpressionAttributeNames'] = {}
+
+        pe = []
+        for attri, attr in enumerate(attrs):
+            name_parts = []
+            for parti, part in enumerate(attr.split(".")):
+                # replace the attrs with expression attributes so we can use reserved names (like count)
+                # convert names like child.sub -> to #pe1_1.#pe1_2
+                pename = '#pe{}'.format('_'.join([str(attri), str(parti)]))
+                self.dynamo_kwargs['ExpressionAttributeNames'][pename] = part
+                name_parts.append(pename)
+            pe.append('.'.join(name_parts))
+
+        self.dynamo_kwargs['ProjectionExpression'] = ', '.join(pe)
+        self._partial = True
+        return self
+
+    def recursive(self):
```

Comment:
I'm not quite following what makes the continuous version recursive - isn't it more of an unbounded iterator?

Reply:
What’s the difference between recursion and unbounded iteration?

Reply:
Rather, what would make more sense to you for a method name/attr?

Reply:
OH. I missed __next__ calling itself

Reply:
I am having trouble reading through this iterator code - I'll have to take a look when I'm feeling more fresh or something